web/src/user_profile.ts

```diff
@@ -584,16 +669,16 @@ function handle_remove_stream_subscription(target_user_id, sub, success, failure
     }
 }
 
-export function show_edit_bot_info_modal(user_id, $container) {
+export function show_edit_bot_info_modal(user_id: number, $container: JQuery): void {
     const bot = people.maybe_get_user_by_id(user_id);
-    const owner_id = bot_data.get(user_id).owner_id;
+    const owner_id = bot_data.get(user_id)!.owner_id!;
     const owner_full_name = people.get_full_name(owner_id);
     const is_active = people.is_person_active(user_id);
 
     if (!bot || !bot_data.get(user_id)) {
         return;
     }
```

Comment:
This logic seems a bit confused; you have asserts that the user ID exists in `bot_data`, and then a check if it's not there. I think the logic is out of order here.

Reply:
Yes, the logic here was wrong, I've fixed it.